core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala

```diff
@@ -118,7 +127,7 @@ private[spark] class LiveListenerBus extends SparkListenerBus with Logging {
    * Log an error message to indicate that the event queue is full. Do this only once.
    */
   private def logQueueFullErrorMessage(): Unit = {
-    if (!queueFullErrorMessageLogged) {
+    if (queueFullErrorMessageLogged.compareAndSet(false, true)) {
```

Comment:
How bad is this race condition? The worst thing that can happen is that we log the same error message multiple times. Is my understanding correct?